hubverse_annotator/app.py

```diff
@@ -0,0 +1,214 @@
+import streamlit as st
+
+PYRENEW_MODELS = {
+    "CFA_Pyrenew-Pyrenew_HE_COVID": "HE_COVID",
+    "CFA_Pyrenew-Pyrenew_H_COVID": "H_COVID",
+    "CFA_Pyrenew-Pyrenew_HW_COVID": "HW_COVID",
+}
+
+
+def create_forecast_chart(
+    hubverse_table: pl.DataFrame, reference_date: str
+) -> alt.Chart:
+    """
+    Ingests a hubverse table (polars) and a reference_date string,
+    pivots quantiles wide, melts back long, then creates a
+    stacked‐area (streamgraph‐style) chart where X is the discrete
+    set of target_end_date weeks.
+    """
+    wide = hubverse_table.pivot(
+        values="value",
+        index=["model", "target_end_date", "reference_date"],
+        on="output_type_id",
+    ).sort(by=["model", "target_end_date", "reference_date"])
+    pdf = wide.to_pandas().reset_index()
+    pdf["target_end_date"] = pd.to_datetime(pdf["target_end_date"])
+    qcols = [
+        c
+        for c in pdf.columns
+        if isinstance(c, float)
+        or (isinstance(c, str) and c.replace(".", "", 1).isdigit())
+    ]
+
+    long_df = pdf.melt(
+        id_vars=["model", "target_end_date", "reference_date"],
+        value_vars=qcols,
+        var_name="quantile",
+        value_name="value",
+    )
+    long_df["date_str"] = long_df["target_end_date"].dt.strftime("%Y-%m-%d")
+    long_df["quantile_num"] = long_df["quantile"].astype(float)
+    long_df["opacity"] = 1 - (long_df["quantile_num"] - 0.5).abs() * 2
+    unique_dates = list(long_df["date_str"].unique())
+    sel = alt.selection_point(fields=["model"], bind="legend")
+    chart = (
+        alt.Chart(long_df)
+        .mark_area(interpolate="linear")
+        .encode(
+            x=alt.X(
+                "date_str:O",
+                title="Target End Date (weekly)",
+                sort=unique_dates,
+                axis=alt.Axis(labelAngle=-45),
+            ),
+            y=alt.Y("value:Q", stack="center", title="Forecast Value"),
+            color=alt.Color(
+                "quantile:O", title="Quantile", scale=alt.Scale(scheme="blues")
+            ),
+            facet=alt.Facet("model:N", columns=1, title=None),
+            opacity=alt.Opacity("opacity:Q", legend=None),
+        )
+        .add_params(sel)
+        .properties(
+            width=600,
+            height=100 * long_df["model"].nunique(),
+        )
+        .interactive()
+    )
+    return chart
+
+
+def main() -> None:
+    # initiate logging
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger(__name__)
+    # record start time
+    start_time = time.time()
+    # begin streamlit application
+    st.title("Forecast Annotator")
+    uploaded_file = st.file_uploader(
+        "Upload Hubverse File", type=["csv", "parquet"]
+    )
+    # load the hubverse data
+    if uploaded_file is not None:
+        if uploaded_file.name.endswith("parquet"):
+            smhub_table = pl.read_parquet(uploaded_file)
+        else:
+            smhub_table = pl.read_csv(uploaded_file)
```

Comment:
check if it's a `csv` before assuming and trying to read it.

Reply:
In theory, this is guaranteed by `st.file_uploader`, but better to be explicit, I think.